File: merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int total = m + n;
        
        int[] temp = new int[total];
        int i = 0, j = 0, curr = 0;
        while (i < m && j < n) {
            if (nums1[i] <= nums2[j]) {
                temp[curr] = nums1[i];
                i++;
            } else {
                temp[curr] = nums2[j];
                j++;
            }
            curr++;
        }
        
        while (i < m) {
            temp[curr] = nums1[i];
            i++;
            curr++;
        }
        
        while (j < n) {
            temp[curr] = nums2[j];
            j++;
            curr++;
        }
        
        for (int k = 0; k < total; k++) {
            nums1[k] = temp[k];
        }
    }
}

File: merge-sorted-array/merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        ptr1 = m - 1
        ptr2 = n - 1
        curr = m + n - 1
        
        while ptr1 >= 0 and ptr2 >= 0:
            if nums1[ptr1] >= nums2[ptr2]:
                nums1[curr] = nums1[ptr1]
                ptr1 -= 1
            else:
                nums1[curr] = nums2[ptr2]
                ptr2 -= 1
            
            curr -= 1
        
        while ptr2 >= 0:
            nums1[curr] = nums2[ptr2]
            ptr2 -= 1
            curr -= 1
        